src/BooksFormCommand.php

```diff
@@ -0,0 +1,198 @@
+<?php
+declare(strict_types = 1);
+
+class BooksFormCommand
+{
+    /**
+     * @var string
+     */
+    private $author;
+    /**
+     * @var string
+     */
+    private $title;
+    /**
+     * @var string
+     */
+    private $genre;
+    /**
+     * @var string
+     */
+    private $price;
+    /**
+     * @var string
+     */
+    private $publishDate;
+    /**
+     * @var string
+     */
+    private $description;
+    /**
+     * @var \TheSeer\fDOM\fDOMDocument
+     */
+    private $dataModel;
+    /**
+     * @var Request
+     */
+    private $postRequest;
+    /**
+     * @var XmlBackend
+     */
+    private $xmlBackend;
+
+    /**
+     * BooksFormCommand constructor.
+     * @param XmlBackend $xmlBackend
+     * @param Request $postRequest
+     * @param \TheSeer\fDOM\fDOMDocument $dataModel
+     */
+    public function __construct(XmlBackend $xmlBackend, Request $postRequest, \TheSeer\fDOM\fDOMDocument $dataModel)
+    {
+        $this->author = $postRequest->getParameter('author');
+        $this->title = $postRequest->getParameter('title');
+        $this->genre = $postRequest->getParameter('genre');
+        $this->price = $postRequest->getParameter('price');
+        $this->publishDate = $postRequest->getParameter('publishDate');
+        $this->description = $postRequest->getParameter('description');
+
+        $this->dataModel = $dataModel;
+        $this->postRequest = $postRequest;
+        $this->xmlBackend = $xmlBackend;
+    }
+
+    public function validateRequest()
+    {
+        try {
+            new Price($this->price);
+        } catch (\InvalidArgumentException $e) {
+            $this->dataModel->queryOne('//field[@name="price"]/error')->nodeValue = 'Bitte geben Sie einen gültigen Preis ein';
+        }
+
+        if ($this->genre != 'Computer' && $this->genre != 'Fantasy' && $this->genre != 'Romance' && $this->genre != 'Horror' && $this->genre != 'Science Fiction') {
```

Comment:
$allowedGenres = array('Foo', 'Bar', 'Baz');
if(!in_array($this->genre, $allowedGenres) {